NikolayGrabenko/src/library/controller/Library.java

```diff
@@ -0,0 +1,298 @@
+        return printedEditionsBase;
+    }
+
+    public ArrayListMy getPrintedEditionsAtReaders() {
+        return printedEditionsAtReaders;
+    }
+
+    public ArrayListMy getPrintedEditionsAvailable() {
+        return printedEditionsAvailable;
+    }
+
+    public ArrayListMy getRegisteredReaders() {
+        return registeredReaders;
+    }
+
+    public ArrayListMy getBlackListOfReaders() {
+        return blackListOfReaders;
+    }
+
+    public Library() {
+        printedEditionsBase = new ArrayListMy();
+        printedEditionsAvailable = new ArrayListMy();
+        printedEditionsAtReaders = new ArrayListMy();
+        registeredReaders = new ArrayListMy();
+        blackListOfReaders = new ArrayListMy();
+
+    }
+
+    public boolean isRegistered(Reader someReader) {
+        boolean isRegistered = false;
+        for (int i = 0; i < registeredReaders.size(); i++) {
+            if (registeredReaders.get(i).equals(someReader)) {
+                isRegistered = true;
+                break;
+            }
+        }
+        return isRegistered;
+    }
+
+    public boolean isInBlackList(Reader reader) {
+        boolean isInBlackList = false;
+        if (reader != null) {
+            for (int i = 0; i < blackListOfReaders.size(); i++) {
+                Reader readerTemp = (Reader) blackListOfReaders.get(i);
+                if (readerTemp.equals(reader)) {
+                    isInBlackList = true;
+                    break;
+                }
+            }
+        }
+        return isInBlackList;
+    }
+
+    public boolean isAvailablePrintEds(PrintedEditions edition) {
+        boolean isAvailablePrintEds = false;
+        if (edition != null) {
+            for (int i = 0; i < printedEditionsAvailable.size(); i++) {
+                PrintedEditions editionTemp = (PrintedEditions) printedEditionsAvailable.get(i);
+                if (editionTemp.getNumberOfCopiesAvailable() > 0) {
+                    isAvailablePrintEds = true;
+                    break;
+                }
+            }
+        }
+        return isAvailablePrintEds;
+    }
+    //    HELP!!!!!
+    public void sortPrintedEditionByComparator(ArrayListMy list, Comparator comparator) {
+        for (int i = 0; i < list.size(); i++) {
+            for (int j = list.size() - 1; j > i; j--) {
+                if (list.get(j) != null) {
+                    if (comparator.compare(list.get(j), list.get(j - 1)) < 0) {
+                        PrintedEditions edition1 = (PrintedEditions) list.get(j);
+                        PrintedEditions edition2 = (PrintedEditions) list.get(j - 1);
+                        PrintedEditions editionTemp;
+                        editionTemp = edition1;
+                        edition1 = edition2;
+                        edition2 = editionTemp;
+                    }
+                }
+            }
+        }
+    }
+//    HELP!!!!!
+    public ArrayListMy sortReadersByComparator(ArrayListMy list, Comparator comparator) {
```

Comment:
public ArrayListMy sortReadersByComparator(ArrayListMy list, Comparator comparator) {
        for (int i = 0; i < list.size(); i++) {
            for (int j = list.size() - 1; j > i; j--) {
                    if (comparator.compare(list.get(j), list.get(j - 1)) < 0) {
                        Reader reader1 = (Reader) list.get(j);
                        Reader reader2 = (Reader) list.get(j - 1);
                        Reader readerTemp;
                        readerTemp = reader1;
                        list.set(j, reader1);
                        list.set(j - 1, readerTemp);
                    }
            }
        }
        return list;
    }